### PR TITLE
Add deepmark CI tests

### PR DESCRIPTION
**Pull request by lnguyen-nvn**
## Description ##

1.NGRAPH-2005: Framework to execute inference tests for  9 topologies . 
2. NGRAPH-1712 :  Exit out before compiling or running tests if we fail to check style (make cpplint)
3. NGRAPH-1837: Update CI framework for merging jobs, daily validation jobs to handle docker images properly, better designs. 
4. Re-write some scripts in order to work on both python 2 and 3. 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: docker/scripts/lib_validation_testing.py [hunk 1]
```diff
@@ -264,103 +259,332 @@ def runBenchmarkScoreScript(script=None,          # Script to run
...
+    cmd = "OMP_NUM_THREADS={} KMP_AFFINITY={} KMP_BLOCKTIME={} {} {} --network inception-v4 --batch-size {}".format(ompNumThreads, kmpAff, kmpBlocktime, python_lib.strip(), script, batchsize.strip())
+    print("The command for deepmark script is: {}".format(cmd))
+    runLog = runCommand(command=cmd, logID=logID)
+    return runLog
+
+# End: def runInceptionV4Script()
+
+
+def runResnet50DeepMarkScript(sourceDir=None,
+                    script=None,          # Script to run
```
Comment (mbrookhart) [inline docker/scripts/lib_validation_testing.py]: This is odd formatting, but not super critical.

**Comment by lnguyen-nvn** [comment]
TESTNOW

**Comment by lnguyen-nvn** [comment]
TESTNOW

**Comment by lnguyen-nvn** [comment]
TESTNOW

**Comment by lnguyen-nvn** [comment]
TESTNOW

**Comment by lnguyen-nvn** [comment]
TESTNOW

**Comment by lnguyen-nvn** [comment]
TESTNOW

**Comment by lnguyen-nvn** [comment]
TESTNOW

**Pull request merged**